.github/workflows/pulsar-ci.yaml

```diff
@@ -25,6 +25,7 @@ on:
   push:
     branches:
       - branch-*
+      - master
```

Comment:
I don't think we should run tests for every commit on the master. If you want to generate a codecov report every day, you may schedule a daily cron job. Anyway, this is a significant change to the CI workflow.

cc @codelipenghui @lhotari

Reply:
IMO, for each PR, we should only generate the codecov report for the new changes. I'm not sure does codecov can support?

Reply:
I don't find any configuration on `report for the new changes` except `require_changes` in `comment` selection and `patch` in `coverage/status section`, but It seems they are not what @codelipenghui wants. Could you help to take a look again?

https://docs.codecov.com/docs/codecovyml-reference